tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -32,22 +39,29 @@ pub use json::*;
 /// signature as `format_event`.
 ///
 /// [`FmtSubscriber`]: ../fmt/struct.Subscriber.html
-pub trait FormatEvent<N> {
+pub trait FormatEvent<S, N>
+where
+    S: Subscriber + for<'a> LookupSpan<'a> + LookupMetadata,
+    N: for<'writer> FormatFields<'writer> + 'static,
```

Comment:
> Also, I think we might want to consider moving the S type parameter from the trait to the format_event method — this would allow implementations of this trait for specific types `S: Subscriber + for<'a> LookupSpan<'a> + LookupMetadata`.

I think I'm onboard with that, but I don't _really_ mind either way.